internal/slicer/report.go

```diff
@@ -30,20 +32,21 @@ type Report struct {
 
 // NewReport returns an empty report for content that will be based at the
 // provided root path.
-func NewReport(root string) *Report {
-	return &Report{
+func NewReport(root string) (*Report, error) {
+	if !filepath.IsAbs(root) {
+		return nil, fmt.Errorf("cannot use relative path for report root: %q", root)
+	}
+	report := &Report{
 		Root:    filepath.Clean(root) + "/",
 		Entries: make(map[string]ReportEntry),
 	}
+	return report, nil
 }
 
 func (r *Report) Add(slice *setup.Slice, fsEntry *fsutil.Entry) error {
-	if !strings.HasPrefix(fsEntry.Path, r.Root) {
-		return fmt.Errorf("cannot add path %q outside of root %q", fsEntry.Path, r.Root)
-	}
-	relPath := filepath.Clean("/" + strings.TrimPrefix(fsEntry.Path, r.Root))
-	if fsEntry.Mode.IsDir() {
-		relPath = relPath + "/"
+	relPath, err := r.sanitizeAbsPath(fsEntry.Path, fsEntry.Mode.IsDir())
+	if err != nil {
+		return fmt.Errorf("cannot add path: %s", err)
```

Comment:
Per last conversation/review, these errors need tuning so we can understand what they're trying to convey, and also so that they are more consistent with each other. For example:

"cannot add path to report: %s"
"cannot mutate path in report: %s"
"cannot mutate path in report: %s not previously reported"
"cannot mutate path in report: %s is a directory"

Reply:
Agree with one minor change `reported -> added` in:
```
cannot mutate path in report: %s not previously added
```
to make it more accurate with the report API of `Add` and `Mutate`.